scripts/import-iidx.mjs

```diff
@@ -0,0 +1,345 @@
+
+        var chartData = []
+        for (let v of chartSlot.entries()) {
+          // DP -> double, SP -> single
+          const chartStyle = ["", "double", "single"]["ZDS".indexOf(v[1][0])];
+          // Map the slot to the full enumeration element
+          const diffClass = (chartStyle != "") ? ["beginner", "normal", "hyper", "another", "leggendaria"]["BNHAL".indexOf(v[1][2])] : "";
+          const chartLevel = chartLevels[v[0]]
+          if (diffClass != "" && chartLevel != 0) {
+            // Chart slot exists for this song?
+            var chartInfo = {
+              style: chartStyle,
+              lvl: chartLevel,
+              diffClass: diffClass
+            }
+            if (chartBPMs[v[0]] != songBPM) {
+              // Per-chart BPM?
+              chartInfo.bpm = chartBPMs[v[0]]
+            }
+            chartData.push(chartInfo)
+          }
+        }
+
+        // Title and subtitle
+        var nameExt = decodeHTML(await unwrapHTML(titletbl[songTag][5]), {scope: "strict"})
+        if (titletbl[songTag][6]) {
+          nameExt += "\n" + decodeHTML(await unwrapHTML(titletbl[songTag][6]), {scope: "strict"})
+        }
+
+        // Unlock category, if applicable
+        var songFlags = []
+        for (let em of eventMap.entries()) {
+          if (em[1][1].includes(songTag)) {
+            songFlags.push(eventFlags.get(eventTags[em[0]]))
+          }
+        }
+
+        // Version of origin (or first AC inclusion)
+        const folderNumber = titletbl[songTag][0] || 0
+        const folderName = folderNames[folderNumber]
+        const folderFile = folderName.replaceAll(" ", "-")
+
+        var songData = {
+          name: nameExt,
+          artist: decodeHTML(titletbl[songTag][4] || "[artist N/A]", {scope: "strict"}),
+          genre: decodeHTML(titletbl[songTag][3] || "[genre N/A]", {scope: "strict"}),
+          flags: songFlags,
+          bpm: songBPM,
+          jacket: `iidx/${folderFile}.svg`,
+          folder: folderName,
+          charts: chartData,
+          saIndex: `${songList.length}`
+        }
+
+        songList.push(songData)
+      }
+      catch (err) {
+        console.warn(`Something's up with song tag ${songTag}:\n${err}`)
+      }
+    }
+  }
+
+  data.songs = songList
+
+  console.log(`Successfully built chart info database using textage JS`);
+
+  console.log(`Building version folder SVG jackets...`);
+  const jacketPath = path.join(__dirname, "../src/assets/jackets/iidx");
+  const jacketTemplate = await fs.readFile(path.resolve(path.join(jacketPath, "_template.svg")), {encoding: "utf-8"});
+  for (let fn of folderNames.entries()) {
+    const folderName = folderNames[fn[0]]
+    const folderFile = folderName.replaceAll(" ", "-")
+    var jacketSpecific = jacketTemplate;
+    for (let jp of jacketPaletteEntries.entries()) {
+      jacketSpecific = jacketSpecific.replaceAll(`{{${jp[1]}}}`, jacketPalettes[fn[0]][jp[0]])
```

Comment:
it actually didn't even cross my mind to use the key art directly! I might leave it up to community feedback and change it to that later. in any case, having svg fallbacks is nice because they're pretty light